File: view/src/main/java/com/senyk/volodymyr/schedulesapp/di/component/AppComponent.java
package com.senyk.volodymyr.schedulesapp.di.component;

import com.senyk.volodymyr.schedulesapp.di.module.ViewModelsModule;
import com.senyk.volodymyr.schedulesapp.view.fragments.base.BaseFragment;

import javax.inject.Inject;
import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = ViewModelsModule.class)
public interface AppComponent {
    void inject(BaseFragment fragment);
}

File: presentation/src/main/java/com/senyk/volodymyr/schedulesapp/di/component/AppComponent.kt
package com.senyk.volodymyr.schedulesapp.di.component

import android.content.Context
import com.senyk.volodymyr.schedulesapp.di.module.ActivitiesContributor
import com.senyk.volodymyr.schedulesapp.presentation.feature.App
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        ActivitiesContributor::class
    ]
)
@Singleton
interface AppComponent : AndroidInjector<App> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun context(context: Context): Builder

        fun build(): AppComponent
    }
}
